tfhe/src/high_level_api/glwe_ct_list.rs

```diff
@@ -0,0 +1,211 @@
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct GlwePackedCompressedCiphertextList(
+    crate::integer::ciphertext::GlwePackedCompressedCiphertextList,
+);
+
+impl Named for GlwePackedCompressedCiphertextList {
+    const NAME: &'static str = "high_level_api::CompactCiphertextList";
+}
+
+impl GlwePackedCompressedCiphertextList {
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    pub fn get_kind_of(&self, index: usize) -> Option<crate::FheTypes> {
+        Some(match self.0.get_kind_of(index)? {
+            DataKind::Unsigned(n) => {
+                let num_bits_per_block = self.0.packed_list.message_modulus.0.ilog2() as usize;
+                let num_bits = n * num_bits_per_block;
+                match num_bits {
+                    2 => crate::FheTypes::Uint2,
+                    4 => crate::FheTypes::Uint4,
+                    6 => crate::FheTypes::Uint6,
+                    8 => crate::FheTypes::Uint8,
+                    10 => crate::FheTypes::Uint10,
+                    12 => crate::FheTypes::Uint12,
+                    14 => crate::FheTypes::Uint14,
+                    16 => crate::FheTypes::Uint16,
+                    32 => crate::FheTypes::Uint32,
+                    64 => crate::FheTypes::Uint64,
+                    128 => crate::FheTypes::Uint128,
+                    160 => crate::FheTypes::Uint160,
+                    256 => crate::FheTypes::Uint256,
+                    _ => return None,
+                }
+            }
+            DataKind::Signed(n) => {
+                let num_bits_per_block = self.0.packed_list.message_modulus.0.ilog2() as usize;
+                let num_bits = n * num_bits_per_block;
+                match num_bits {
+                    2 => crate::FheTypes::Int2,
+                    4 => crate::FheTypes::Int4,
+                    6 => crate::FheTypes::Int6,
+                    8 => crate::FheTypes::Int8,
+                    10 => crate::FheTypes::Int10,
+                    12 => crate::FheTypes::Int12,
+                    14 => crate::FheTypes::Int14,
+                    16 => crate::FheTypes::Int16,
+                    32 => crate::FheTypes::Int32,
+                    64 => crate::FheTypes::Int64,
+                    128 => crate::FheTypes::Int128,
+                    160 => crate::FheTypes::Int160,
+                    256 => crate::FheTypes::Int256,
+                    _ => return None,
+                }
+            }
+            DataKind::Boolean => crate::FheTypes::Bool,
+        })
+    }
+
+    pub fn get<T>(&self, index: usize) -> Option<crate::Result<T>>
+    where
+        T: Expandable,
+    {
+        with_cpu_internal_keys(|key| {
+            self.0.get(
+                index,
+                key.decompression_key
+                    .as_ref()
+                    .expect("Decompression key not set in server key"),
+            )
+        })
```

Comment:
Same idea as the `build` method of the builder, maybe the panic cases could be put into the `Err` of the returned `crate::Result`

Reply:
Done